src/electrons.cxx

```diff
@@ -112,6 +116,239 @@ PtCorrectionMC(ROOT::RDF::RNode df,
+                throw std::runtime_error("unknown variation");
+            }
+
+            // logging output
+            Logger::get("physicsobject::electron::PtCorrectionMC")
+                ->debug("ele pt before {}, ele pt after {}, sf {}, variation {}",
+                        pt.at(i), pt_corrected.at(i), sf, variation);
+
+        }
+
+        return pt_corrected;
+    };
+
+    return df.Define(
+        outputname,
+        correction,
+        {pt, eta, delta_eta_sc, r9, event_seed}
+    );
+}
+
+/**
+ * @brief This function applies energy scale corrections to data. The corrections are
+ * obtained from a dedicated correctionlib file.
+ * 
+ * For Run 3 samples, the electron scale correction is not available in the NanoAOD files
+ * and the corrections have to be evaluated using a centrally provided correctionlib file.
+ * This function should only be used  The documentation of the file content
+ * can be found here:
+ * 
+ * - [2022preEE](https://cms-nanoaod-integration.web.cern.ch/commonJSONSFs/summaries/EGM_2022_Summer22_electronSS_EtDependent.html)
+ * - [2022postEE](https://cms-nanoaod-integration.web.cern.ch/commonJSONSFs/summaries/EGM_2022_Summer22EE_electronSS_EtDependent.html)
+ * - [2023preBPix](https://cms-nanoaod-integration.web.cern.ch/commonJSONSFs/summaries/EGM_2023_Summer23_electronSS_EtDependent.html)
+ * - [2023postBPix](https://cms-nanoaod-integration.web.cern.ch/commonJSONSFs/summaries/EGM_2023_Summer23BPix_electronSS_EtDependent.html)
+ *
+ * An implementation recipe is provided here:
+ * [egmScaleAndSmearingExample.py](https://gitlab.cern.ch/cms-nanoAOD/jsonpog-integration/-/blob/master/examples/egmScaleAndSmearingExample.py).
+ * 
+ * @param df input dataframe
+ * @param correction_manager correction manager responsible for loading the
+ * correction scale uncertainty patch file
+ * @param outputname name of the output column for corrected \f$p_T\f$ values
+ * @param pt name of the column containing electron \f$p_T\f$ values
+ * @param eta name of the column containing electron pseudorapidities
+ * @param gain name of the column containing electron gain values
+ * @param r9 name of the column containing the R9 value of the electron's
+ * supercluster
+ * @param run name of the column containing the run number
+ * @param sf_file path to the correction file for the energy scale corrections
+ * and variations
+ * @param variation name of the energy correction variation that should be
+ * calculated (e.g., "resolutionUp", "resolutionDown", "scaleUp", "scaleDown"),
+ * for "nominal" nothing is done because energy correction is already applied
+ *
+ * @return a dataframe containing the varied electron transverse momenta
+ * 
+ * @note This function is intended for analyses working with Run 3 NanoAODv12
+ * or higher. In the Run 2 NanoAODv12 samples, the scale correction in data
+ * is already applied in the NanoAOD files. Look at
+ * ``physicsobject::electron::PtCorrectionMC`` for running on Run 2 samples.
```

Comment:
Looking at `PtCorrectionMC` is not correct in this case, right? I would just drop it.